qp_shotgun/shogun/shogun.py

```diff
@@ -0,0 +1,189 @@
+    cmds.append(
+        'shogun functional '
+        '--database {database} '
+        '--input {input} '
+        '--output {output} '
+        '--level {level}'.format(
+            database=parameters['database'],
+            input=profile_dir,
+            output=join(temp_dir, 'profile.functional.%s.tsv' % sel_level),
+            level=sel_level))
+
+    return cmds
+
+
+def generate_biom_conversion_commands(input_fp, output_dir, level):
+    cmds = []
+    cmds.append(
+        'biom convert -i {input} '
+        '-o {output} '
+        '--table-type="OTU table" '
+        '--process-obs-metadata taxonomy --to-hdf5'.format(
+            input=input_fp,
+            output=join(output_dir, 'otu_table.%s.biom' % level)))
+
+    return cmds
+
+
+def shogun(qclient, job_id, parameters, out_dir):
+    """Run Shogun with the given parameters
+
+    Parameters
+    ----------
+    qclient : tgp.qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to run split libraries
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    bool, list, str
+        The results of the job
+    """
+    # Step 1 get the rest of the information need to run Atropos
+    qclient.update_job_step(job_id, "Step 1 of 6: Collecting information")
+    artifact_id = parameters['input']
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fps = artifact_info['files']
+
+    # Get the artifact metadata
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    qiime_map = prep_info['qiime-map']
+
+    # Step 2 converting to fna
+    qclient.update_job_step(
+        job_id, "Step 2 of 6: Converting to FNA for Shogun")
+
+    with TemporaryDirectory(dir=out_dir, prefix='shogun_') as temp_dir:
+        rs = fps['raw_reverse_seqs'] if 'raw_reverse_seqs' in fps else []
+        samples = make_read_pairs_per_sample(
+            fps['raw_forward_seqs'], rs, qiime_map)
+
+        comb_fp = generate_fna_file(temp_dir, samples)
+        # Combining files
+        parameters = _format_params(parameters, SHOGUN_PARAMS)
+        # Step 3 align
+        qclient.update_job_step(
+            job_id, "Step 3 of 6: Aligning FNA with Shogun")
+        generate_shogun_align_commands(
+            comb_fp, temp_dir, parameters)
+        # Step 4 taxonomic profile
+        qclient.update_job_step(
+            job_id, "Step 4 of 6: Taxonomic profile with Shogun")
+        cmd, profile_fp = generate_shogun_assign_taxonomy_commands(
+            temp_dir, parameters)
+        # Step 5 functional profile
+        qclient.update_job_step(
+            job_id, "Step 5 of 6: Functional profile with Shogun")
+        levels = ['genus', 'species', 'level']
+        for level in levels:
```

Comment:
You'll also want to create redistributed Biom files for the taxonomy tables at these levels, a la: 

https://github.com/biocore/oecophylla/blob/251fa6c5a34014b919736acf3e003da13f09053b/oecophylla/taxonomy/taxonomy.rule#L378